projecttacoma/bulk-export-server
# Add docker setup for bulk export

# Summary
This PR sets up a working Docker environment for this server and changes some things so that it can be run on our abacus-dev ECE.

## New behavior
Server can be run with Docker! Also pushed this repository to docker hub and added it to abacus-dev.

## Code changes
- `.env` - adds `PUBLIC_BULK_SERVER` variable and changes `PORT` to 3000 for consistency
- `.env.test` - adds `PUBLIC_BULK_SERVER` variable
- `Dockerfile` - upgrades node, adds `ENV HOST` and `ENV PORT` variables
- `docker-compose.example.yml` - example docker-compose for future reference (similar to that in the `measure-repository`)
- `docker-compose.yml` - add redis, upgrade mongo, add `PUBLIC_BULK_SERVER`
- `postTransactionBundles.js`/`bulkstatus.service.js`/`export.service.js`/`bulkstatus.service.test.js` - use `PUBLIC_BULK_SERVER`

# Testing guidance
- `npm run check`
- Start up the server with docker with `docker compose up --build`
- Use Insomnia to ensure functionality is the same as if the server was spun up locally. See PR #39 for example Insomnia requests to get started.
- Start the server locally to ensure that all runs the same.
